domain/application/state/application_test.go

```diff
@@ -3028,6 +3030,267 @@ func (s *applicationStateSuite) TestHashConfigAndSettings(c *gc.C) {
 	}
 }
 
+func (s *applicationStateSuite) TestConstraintFull(c *gc.C) {
+	id := s.createApplication(c, "foo", life.Alive)
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		addConstraintStmt := `INSERT INTO "constraint" (uuid, arch, cpu_cores, cpu_power, mem, root_disk, root_disk_source, instance_role, instance_type, container_type_id, virt_type, allocate_public_ip, image_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+		_, err := tx.ExecContext(ctx, addConstraintStmt, "constraint-uuid", "amd64", 2, 42, 8, 256, "root-disk-source", "instance-role", "instance-type", 1, "virt-type", true, "image-id")
+		if err != nil {
+			return err
+		}
```

Comment:
Knit: Maybe use a `Txn` so you can take advantage of sqlair here?

Reply:
I prefer to keep the same pattern used all over this same test file, at least it's consistent.